akka-http/src/main/scala/akka/http/scaladsl/coding/Decoder.scala

```diff
@@ -10,18 +10,21 @@ import akka.stream.{ FlowShape, Materializer }
 import akka.stream.stage.GraphStage
 import akka.util.ByteString
 import headers.HttpEncoding
-import akka.stream.scaladsl.{ Sink, Source, Flow }
+import akka.stream.scaladsl.{ Flow, Sink, Source }
 
 import scala.concurrent.Future
 
 trait Decoder {
   def encoding: HttpEncoding
 
-  def decode[T <: HttpMessage](message: T)(implicit mapper: DataMapper[T]): T#Self =
+  def decodeMessage(message: HttpMessage): HttpMessage#Self =
     if (message.headers exists Encoder.isContentEncodingHeader)
-      decodeData(message).withHeaders(message.headers filterNot Encoder.isContentEncodingHeader)
+      message.transformEntityDataBytes(decoderFlow).withHeaders(message.headers filterNot Encoder.isContentEncodingHeader)
     else message.self
 
+  def decode[T <: HttpMessage](message: T)(implicit mapper: DataMapper[T]): T#Self =
```

Comment:
I think we can now deprecate this old type signature in favor of `decodeMessage`. Could you add the deprecation annotation?